packages/core/http/core-http-server-internal/src/http_server.ts

```diff
@@ -385,6 +389,44 @@ export class HttpServer {
     }
   }
 
+  private getDeprecatedRoutes(): RouterDeprecatedRouteDetails[] {
```

Comment:
I see this currently gets called every time our deprecations REST API gets called... I wonder if we should cache using something like `_.once`?

Reply:
I've written [some docs about this in the public](https://github.com/elastic/kibana/pull/196081/files#diff-5244a4eb2bd1f47aadae36cc27d80a99e9ed7e94631ebe5c94688056b0727285R365-R370) interface for this function

>   * Provides a list of all registered deprecated routes {{@link RouterDeprecatedRouteDetails | information}}.
>   * The routers will be evaluated everytime this function gets called to
>   * accommodate for any late route registrations
>   * @returns {RouterDeprecatedRouteDetails[]}